base_layer/core/src/proof_of_work/lwma_diff.rs

```diff
@@ -143,6 +147,14 @@ impl LinearWeightedMovingAverage {
         }
         self.target_difficulties.push((timestamp, target_difficulty));
     }
+
+    pub fn update_consensus_constants(&mut self, constants: &ConsensusConstants, pow_algo: PowAlgorithm) {
```

Comment:
Exposing this method here seems like a wrong coupling. It suggests an order that methods in this class may be called in.